xcell/mappers/mapper_2MPZ.py

```diff
@@ -0,0 +1,152 @@
+        self.mask = self.get_mask()
+        ipix = hp.ang2pix(self.nside, cat['SUPRA'],
+                          cat['SUPDEC'], lonlat=True)
+        # Mask is binary, so 0.1 or 0.00001 doesn't really matter.
+        return cat[self.mask[ipix] > 0.1]
+
+    def _bin_z(self, cat):
+        return cat[(cat['ZPHOTO'] >= self.z_edges[0]) &
+                   (cat['ZPHOTO'] < self.z_edges[1])]
+
+    def _get_specsample(self, cat):
+        ids = cat['ZSPEC'] > -1
+        return cat[ids]
+
+    def get_nz(self, dz=0, return_jk_error=False):
+        if self.dndz is None:
+            f_exists, f_name = self._check_rerun_file_exists('nz_2MPZ.npz')
+            # Read from file if it exists
+            if f_exists:
+                d = np.load(f_name)
+                zm = d['z_mid']
+                nz = d['nz']
+                nz_jk = d['nz_jk']
+            else:  # Else compute DIR N(z) and jackknife resamples
+                c_p = self.get_catalog()
+                c_s = self._get_specsample(c_p)
+                # Sort spec sample by nested pixel index so jackknife
+                # samples are spatially correlated.
+                ip_s = hp.ring2nest(self.nside,
+                                    hp.ang2pix(self.nside, c_s['SUPRA'],
+                                               c_s['SUPDEC'], lonlat=True))
+                idsort = np.argsort(ip_s)
+                c_s = c_s[idsort]
+                # Compute DIR N(z)
+                z, nz, nz_jk = get_DIR_Nz(c_s, c_p,
+                                          ['JCORR', 'KCORR', 'HCORR',
+                                           'W1MCORR', 'W2MCORR',
+                                           'BCALCORR', 'RCALCORR', 'ICALCORR'],
+                                          zflag='ZSPEC',
+                                          zrange=[0, 1.],
+                                          nz=100,
+                                          njk=self.config.get('n_jk_dir', 100))
+                zm = 0.5*(z[1:] + z[:-1])
+                if f_name is not None:
+                    np.savez(f_name, z_mid=zm, nz=nz, nz_jk=nz_jk)
+            self.dndz = (zm, nz, nz_jk)
+
+        z, nz, nz_jk = self.dndz
+        z_dz = z + dz
+        sel = z_dz >= 0
+        if return_jk_error:
+            njk = len(nz_jk)
+            enz = np.std(nz_jk, axis=0)*np.sqrt((njk-1)**2/njk)
+            return np.array([z_dz[sel], nz[sel], enz[sel]])
+        else:
+            return np.array([z_dz[sel], nz[sel]])
+
+    def get_signal_map(self, apply_galactic_correction=True):
+        if self.delta_map is None:
+            d = np.zeros(self.npix)
+            self.cat_data = self.get_catalog()
+            self.mask = self.get_mask()
+            nmap_data = get_map_from_points(self.cat_data, self.nside,
+                                            ra_name='SUPRA',
+                                            dec_name='SUPDEC')
+            mean_n = np.sum(self.mask*nmap_data)/np.sum(self.mask)
```

Comment:
np.average(nmap_data, weights=self.mask) in case you want to do it with less code

Reply:
Done!